client/patchman-client

```diff
@@ -191,23 +191,14 @@ get_installed_deb_packages() {
         if ${verbose} ; then
             echo 'Finding installed debs...'
         fi
-        OLDIFS=${IFS}
-        IFS="
-"
-        dpkg_query_output=$(dpkg-query -W --showformat="\${Status}\|\${Package} \${Version} \${Architecture}\n" \
-                            | egrep '^(install)|(hold) ok installed' \
-                            | sed -e 's/^\(install\|hold\) ok installed|//g')
```

Comment:
Could we use another sed expression here instead of perl? Just don't want to introduce a dependency that didn't already exist - trying to keep it very minimal.

I got this far but it still needs some work: `-r "s/(.*) ([0-9]*):?(.*)[-+](.*)? (.*)$/'\1' '\2' '\3' '\4' '\5' 'deb'/g"`

Reply:
I don't think we can get a solution to split `version` and `release` as we need to split on the last `-` and I cannot get this to work:
```bash
echo -e "1-2-3\n1.2.3" | sed -E 's/^(.*)(-.*)?/vers:\1 release:\2/'
vers:1-2-3 release:
vers:1.2.3 release:
```
If a solution to this is found using sed, something anlong the lines of this should work:
```
from='^(install|hold) ok installed\|([^ ]*) (([^: ]*):)?(.*)(FIXME_version) (.*)'
to="'\2' '\4' '\5' '\6' '\7' 'deb'"
dpkg-query -W --showformat="\${Status}\|\${Package} \${Version} \${Architecture}\n" |
    sed -nE "s/${from}/${to}/pg" >> "${tmpfile_pkg}"

An alternative is using `awk` (but perl was IMO more readable):
```
        q="\"'\""
        dpkg-query -W --showformat="\${Status}\|\${Package} \${Version} \${Architecture}\n" |
            awk '/^(install|hold) ok installed\|/{
    split($0, status, "|");
    split(status[2], detail, " ");
    n = split(detail[2], epoch_t, ":");
    if (n == 1) {
        version = detail[2];
        epoch = "";
    } else {
        version = epoch_t[2];
        epoch = epoch_t[1];
    }
    n = split(version, version_t, "-")
    if (n == 1) {
        release = "";
    } else {
        release = version_t[n]
        version = substr(version, 1, length(version) - length(release) - 1);
    }
    print '$q' detail[1] '$q', '$q' epoch '$q', '$q' version '$q', '$q' release '$q','$q' detail[3] '$q', '$q'"deb"'$q'
}' >> "${tmpfile_pkg}"
```

Reply:
The perl version is nice and short but I think we should go with the awk version just to keep dependencies to a minimum!

Reply:
I cleaned up the awk version a bit (using a here-document to remove quoting issues, naming all variables, using `|` as separator to `dpkg-query` output to avoid a `split` call) and updated this PR.

As the whole script is indented with spaces, I did not indent `AWK` here-document termination mark (which can be indented only using tabs and with `<<-'AWK'` start mark).

And after (re-)reading `deb-version` manpage, I added support for colons in versions (which are allowed only when there is an epoch so I think they are hardly ever used).

Reply:
@David-5-1 thanks again for your work! Merging

Reply:
No problems! If you ever wanted to untangle the unwieldy awk that's in there for rhel/centos, be my guest!